samples/go/blob-get/blob_get.go

```diff
@@ -51,24 +51,32 @@ func main() {
 	}
 
 	filePath := flag.Arg(1)
+	var file *os.File
+	showProgress := true
 	if filePath == "" {
-		d.CheckErrorNoUsage(errors.New("Empty file path"))
+		file = os.Stdout
```

Comment:
The only real problem here is non-printable characters. Best case they look super weird in bash, worst case they're control characters that do weird things.

I think you might want `unicode.IsGraphic`: https://golang.org/pkg/unicode/#IsGraphic, but it's not that simple, because io.Writer operates in bytes not runes. I had a look and didn't find anything pre-built that does this.

Hacky idea:
```go
// (In the case of writing to stdout... and fixing threading issues...
// and fitting into any other code... and fixing compilation...)
rdr := bufio.NewReader(buffer.Reader())

go func() {
  wtr := bufio.NewWriter(file)
  defer wtr.Close()

  for {
    r, _, err := rdr.ReadRune()
    if err != nil {
      break
    }
    if !unicode.IsGraphic(r) {
      r = '~'
    }
    wtr.WriteRune(r)
  }
}()
```